ugvc/mrd/featuremap_consensus_utils.py

```diff
@@ -0,0 +1,211 @@
+        FeatureMapFields.X_SCORE.value,
+        FeatureMapFields.X_EDIST.value,
+        FeatureMapFields.X_LENGTH.value,
+        FeatureMapFields.X_MAPQ.value,
+        FeatureMapFields.X_INDEX.value,
+        FeatureMapFields.X_FC1.value,
+        FeatureMapFields.X_FC2.value,
+        FeatureMapFields.MAX_SOFTCLIP_LENGTH.value,
+        FeatureMapFields.X_FLAGS.value,
+        # "a3",
+        # "rq"
+    ],
+    "string_list_fields": [
+        FeatureMapFields.X_RN.value,
+        FeatureMapFields.X_CIGAR.value,
+        "rq"
+        # FeatureMapFields.IS_FORWARD.value,
+        # FeatureMapFields.IS_DUPLICATE.value,
+        # "tm"
+    ],
+    "fields_to_write_once": [
+        FeatureMapFields.READ_COUNT.value,
+        FeatureMapFields.TRINUC_CONTEXT_WITH_ALT.value,
+        FeatureMapFields.HMER_CONTEXT_REF.value,
+        FeatureMapFields.HMER_CONTEXT_ALT.value,
+        # FeatureMapFields.IS_CYCLE_SKIP.value,
+        FeatureMapFields.PREV_1.value,
+        FeatureMapFields.PREV_2.value,
+        FeatureMapFields.PREV_3.value,
+        FeatureMapFields.NEXT_1.value,
+        FeatureMapFields.NEXT_2.value,
+        FeatureMapFields.NEXT_3.value,
+    ],
+}
+
+
+def write_a_pileup_record(
+    record_dict: dict,
+    rec_id: str,
+    out_fh: pysam.VariantFile,
+    header: pysam.VariantHeader,
+    min_qual: int,
+    qual_agg_func: callable = np.max,
+    sample_name: str = "SAMPLE",
+):
+    """
+    Write a pileup record to a vcf file
+
+    Inputs:
+        record_dict (dict): A dictionary containing the record information
+        rec_id (str): The record id
+        out_fh (pysam.VariantFile): A pysam file handle of the output vcf file
+        header (pysam.VariantHeader): A pysam VarianyHeader pbject of the output vcf file
+        min_qual (int): The minimum quality threshold
+        qual_agg_func (function): The function to aggregate the quality scores (default: np.max)
+        sample_name (str): The name of the sample (default: "SAMPLE")
+    Output:
+        rec (pysam.VariantRecord): A pysam VariantRecord object
+
+    """
+    format_fields = [GT, AD, DP, VAF]
+
+    rec = header.new_record()
+    rec.chrom = rec_id[0]
+    rec.pos = rec_id[1]
+    rec.ref = rec_id[2]
+    rec.alts = rec_id[3]
+    rec.id = "."
+    rec.qual = qual_agg_func(record_dict[FeatureMapFields.X_QUAL.value])
+    # INFO fields to aggregate
+    # exceptions: X_QUAL
+    rec.info[FeatureMapFields.X_QUAL.value] = list(record_dict[FeatureMapFields.X_QUAL.value])
+    for field in fields_to_collect["numeric_array_fields"]:
+        if field in record_dict:
+            rec.info[field] = list(record_dict[field])
+    for field in fields_to_collect["string_list_fields"]:
+        if field in record_dict:
+            rec.info[field] = "|".join(record_dict[field])
+
+    # FORMAT fields to aggregate
+    record_dict[VAF] = record_dict[FeatureMapFields.FILTERED_COUNT.value] / record_dict[DP]
+    record_dict[GT] = (0, 1)
```

Comment:
Why set the genotype? I guess if there's no other choice a HET variant makes the most sense but isn't it better not to set it at all?